From b2f0d6c0ad5b15690482bcd68c884fbced88b929 Mon Sep 17 00:00:00 2001
From: S-trace <S-trace@list.ru>
Date: Mon, 13 Mar 2017 19:26:52 +0300
Subject: [PATCH 19/20] [LOGS] Advanced log copy routine

---
 gui/action.cpp | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/gui/action.cpp b/gui/action.cpp
index 9b7c034..450d778 100644
--- a/gui/action.cpp
+++ b/gui/action.cpp
@@ -640,10 +640,10 @@ int GUIAction::restoredefaultsettings(std::string arg __unused)
 
 int GUIAction::copylog(std::string arg __unused)
 {
-	operation_start("Copy Log");
+	operation_start("Copy Logs");
 	if (!simulate)
 	{
-		string dst, curr_storage;
+		string dst, curr_storage, date=TWFunc::Get_Current_Date(), cmd;
 		int copy_kernel_log = 0;
 
 		DataManager::GetValue("tw_include_kernel_log", copy_kernel_log);
@@ -656,6 +656,16 @@ int GUIAction::copylog(std::string arg __unused)
 			TWFunc::copy_kernel_log(curr_storage);
 		sync();
 		gui_msg(Msg("copy_log=Copied recovery log to {1}")(dst));
+		dst = DataManager::GetCurrentStoragePath() + "/pmem_"+ date + ".bin.gz";
+		cmd="viewmem 0x7f200000 0x100000 | pigz -9 > " + dst;
+		TWFunc::Exec_Cmd(cmd.c_str() );
+		sync();
+		gui_print("Dumped pmem to %s.\n", dst.c_str());
+		dst = DataManager::GetCurrentStoragePath() + "/last_kmsg_" + date + ".txt.gz";
+		cmd = "if [ -f /proc/last_kmsg ]; then pigz < /proc/last_kmsg > " + dst + "; fi";
+		TWFunc::Exec_Cmd( cmd.c_str() );
+		sync();
+		gui_print("Dumped last_kmsg to %s.\n", dst.c_str());
 	} else
 		simulate_progress_bar();
 	operation_end(0);
-- 
2.10.2

